src/main/Elevator.java

```diff
@@ -211,12 +213,16 @@ public void run() {
 		    elevatorTimer.start();
 			while (this.getIsDoorsOpen()) {};
 			if (this.getDidTimeout()) {
+				this.error = fe.getErrorCode();
+				System.out.println(Thread.currentThread().getName() + " is in error " + this.error + " state");
 				closeDoorsTask.cancel();
 				timer.cancel();
 				this.setDidTimeout(false);
 				fe.setErrorCode(0);
 			} else {
+				this.error = 0;
```

Comment:
should we just set it to fe.getErrorCode() here?

Reply:
fixed